docs/user-guide/a1-01-matrix-layout.md

```diff
@@ -5,24 +5,48 @@ layout: user-guide
+
+* `mul(v, m)` - v is interpretted as a row vector 
+* `mul(m, v)` - v is interpretted as a column vector. 
 
-If the shader code writes `mul(m, v)`, then the last **column** of `m` defines the translation terms. If we use row-major matrix layout, then the host code should make sure the translation terms are filled in at `m + 4, 7, 11` locations in memory.
+Through this mechanism a developer is able to write transforms in their preferred style. 
 
-Alternatively, if the shader code writes `mul(v, m)`, then the last **row** of `m` defines the translation terms. When using row-major matrix layout, the host code should make sure the translation terms are filled in at `m + 12, 13, 14` locations in memory.
+These two styles are not directly interchangable - for a given `v` and `m` then generally `mul(v, m) != mul(m, v)`. For that the matrix needs to be transposed so 
 
-By default, Slang assumes all matrices to be in **row-major** layout, since this is the most nature layout to work with in CPU code: each row of the matrix occupies contiguous space in memory. A user should stick to one of the above practices to get correct result. Note that this is different from `fxc` which assumes `column_major` layout by default. As an example, if the host code uses `glm` library to generate transform matrices, the translation terms will be stored in `[12], [13], [14]` locations in memory. Therefore, the shader code should stick to the `mul(v,m)` convention to ensure correctness.
+* `mul(v, m) == mul(transpose(m), v)`
+* `mul(m, v) == mul(v, transpose(m))`
 
-Slang automatically handles the convention differences when cross-compiling code to GLSL. For example, a `float3x4` matrix will be translated to `mat4x3` in the resulting GLSL. Correspondingly, `mul(v, m)` will be translated to `m*v` in GLSL. Therefore, as long as the user is sticking to the above practices consistently, they will get correct result with the same matrix value in memory regardless of what graphics API they are actually using.
+This behavior is *independent* of how a matrix layout in memory. Host code needs to be aware of how a shader code will interpret a matrix stored in memory, it's layout, as well as the vector interpretation convention used in shader code (ie `mul(v,m)` or `mul(m, v)`).
+
+[Matrix layout](https://en.wikipedia.org/wiki/Row-_and_column-major_order) can be either `row-major` or `column-major`. The difference just determines which elements are contiguous in memory. `Row-major` means the rows elements are contiguous. `Column-major` means the column elements are contiguous.
+
+Another way to think about this difference is in terms of where translation terms should be placed in memory when filling a typical 4x4 transform matrix. For `row-major` matrix layout, translation will be at `m + 12, 13, 14`. For a `column-major` matrix layout, translation will be at `m + 3, 7, 11`.
```

Comment:
This paragraph is not accurate. `row-major` or `column-major` by itself does not dictate where translation terms are stored, it must be combined with whether you are sticking to row-vector or column-vector convention. When using row-vector convention (writing `v*m` in shader code) and using `row-major` matrix layout, then translation terms are stored in 12,13,14. Fliping either one of the math convention or layout majorness flips the matrix storage.

Reply:
True. That the original idea loses some of it's appeal when having to make the distinction. Added row vector.